src/providers/twitch/TwitchChannel.cpp

```diff
@@ -801,13 +804,39 @@ void TwitchChannel::loadRecentMessages()
                 }
             }
 
-            postToThread(
-                [shared, messages = std::move(allBuiltMessages)]() mutable {
-                    shared->addMessagesAtStart(messages);
-                });
+            postToThread([this, shared, root,
+                          messages = std::move(allBuiltMessages)]() mutable {
+                shared->addMessagesAtStart(messages);
+
+                // Notify user about a possible gap in logs
+                if (QString errorCode = root.value("error_code").toString();
+                    !errorCode.isEmpty())
+                {
+                    qCDebug(chatterinoTwitch)
+                        << QString("rm error_code=%1, channel=%2")
+                               .arg(errorCode, this->getName());
+                    if (errorCode == "channel_not_joined")
+                    {
+                        shared->addMessage(makeSystemMessage(
+                            "Recent-messages API isn't currently joined to "
+                            "this channel (in progress or due to an issue)."));
+                    }
```

Comment:
If the `errorCode` is `channel_not_joined` and `messages` is not empty we can deduce that the bot was joined previously but has disconnected, we can inform the user that there may be gaps in the message history: `Message history service recovering, there may be gaps in the message history`

If the `errorCode` is `channel_not_joined` and `messages` is empty I don't think we should print anything, because there's no difference for you vs for the next user who joins and sees nothing other than "Wow! You were the first user here with Chatterino! Congratulations! :rocket:"

Reply:
Should be resolved in 3f1d81f1, let me know if it's good enough.